src/jit/codegencommon.cpp

```diff
@@ -10649,8 +10737,11 @@ void CodeGen::genSetScopeInfo(unsigned       which,
         noway_assert(offset < stkArgSize);
         offset = stkArgSize - offset;
 
-        varLoc->vlType                   = VLT_FIXED_VA;
-        varLoc->vlFixedVarArg.vlfvOffset = offset;
+        newVarLoc                          = *varLoc;
+        newVarLoc.vlType                   = VLT_FIXED_VA;
+        newVarLoc.vlFixedVarArg.vlfvOffset = offset;
```

Comment:
It seems to me like this information should be preserved to the caller, having a copy on the local stack makes this even more confusing to understand than before.